File: PolicyPilot_Deployed/frontend/Policy Pilot P2/Services/UserRuntimeContext.cs
namespace Policy_Pilot_P2.Services;

public sealed class UserRuntimeContext
{
    public string CurrentDateTimeLocal { get; set; } = string.Empty;

    public string CurrentDateIsoUtc { get; set; } = string.Empty;

    public string TimeZone { get; set; } = string.Empty;

    public string Location { get; set; } = "Lubbock";
}

File: PolicyPilot_Deployed/frontend/Policy Pilot P2/wwwroot/userRuntimeContext.js
window.userRuntimeContext = {
    get: function () {
        const now = new Date();

        const pad = (value) => String(value).padStart(2, "0");
        const localDateTime = `${now.getFullYear()}-${pad(now.getMonth() + 1)}-${pad(now.getDate())} ${pad(now.getHours())}:${pad(now.getMinutes())}:${pad(now.getSeconds())}`;

        return {
            currentDateTimeLocal: localDateTime,
            currentDateIsoUtc: now.toISOString(),
            timeZone: Intl.DateTimeFormat().resolvedOptions().timeZone || "Unknown",
            location: "Lubbock"
        };
    }
};
